### ISSUE TITLE
Can't connect to local devices.

### ISSUE THREAD

**Issue by StefanoFranchini** [wg-easy/wg-easy#626]
Hi,

I installed wg-easy on my Synology NAS DS920+ running DSM v7.1.1-42962 following a guide (I'm not sure I can post the link).

I first installed WireGuard for the right system architecture of my NAS, and it is running with no problems:

<img width="502" alt="image" src="https://github.com/wg-easy/wg-easy/assets/70096449/f39cb960-30da-4224-888a-cb81ddd7c984">

and then I installed wg-easy with the default configurations:

```
docker run -d \
  --name=wg-easy \
  -e WG_HOST= my_ip \
  -e PASSWORD= my_pwd \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  weejewel/wg-easy
```

Lastly, I opened the port on my router.

<img width="749" alt="image" src="https://github.com/wg-easy/wg-easy/assets/70096449/589c5fe7-f474-4934-8edf-af41d49bb9d9">

everything went good, and I could access the UI page and create a new client. I then Installed WireGuard app on my iPhone and set-up via qr code.

Now, I can connect without any problem, and I see traffic is passing by:

<img width="855" alt="image" src="https://github.com/wg-easy/wg-easy/assets/70096449/026a8ab8-7be1-4cd3-8262-24d3d6ceaa5d">

However, I can only navigate the web (public, internet) but I can't access my local network.

Please note that my local network has the following IPs: 10.8.0.x, which also is the default IP that the server assigns. for that reason I also tried different configuration, for example:

```
 -e WG_DEFAULT_DNS='8.8.8.8, 8.8.4.4' \
 -e WG_DEFAULT_ADDRESS=192.168.1.x \
```

or

`-e WG_ALLOWED_IPS=10.8.0.0/24 \`
(a strange behaviour with this conf, is that if I use it and check my public IP on my client, it is different than the one of my server - it's the public IP of my mobile ISP -. But if this config is not active, the public ip of my client is always the same as the one of my server)

and again

```
-e WG_POST_UP='iptables -t nat -A POSTROUTING -s subnet of clients -o eth0 -j MASQUERADE' \
-e WG_POST_DOWN='iptables -t nat -D POSTROUTING -s subnet of clients -o eth0 -j MASQUERADE' \
```

but nothing worked.

An other thing that I've noticed, is that even having the following conf:

```
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
```

my /etc/sysctl.conf file did not change, so I added the values by myself, and now it looks like that:

```
kernel.panic = 3
net.core.somaxconn = 65535
net.ipv4.tcp_tw_reuse  = 1
fs.protected_hardlinks = 1
fs.protected_symlinks = 1
kernel.syno_forbid_console=0
kernel.syno_forbid_usb=0
net.ipv6.conf.default.accept_ra_defrtr=0
net.ipv4.conf.default.accept_redirects=0
net.ipv6.conf.default.accept_redirects=0
net.ipv4.conf.default.send_redirects=0
net.ipv4.conf.default.secure_redirects=0
net.ipv6.conf.default.accept_ra=0
net.ipv4.ip_forward=1
net.ipv4.conf.all.src_valid_mark=1
```

After making those changes, I ran:

`sysctl --system`

to make the configs update.

Do you have any clue on what could be my problem?

Thanks

**Comment by eandersons** [comment]
> I can't access my local network

Did you try to access local network instances by IP address or by hostname/domain name?
Do you have your own DNS server?
What exactly you tried, what was the outcome and what is the expected outcome?


> `-e WG_ALLOWED_IPS=10.8.0.0/24`

This means, that only the traffic for 10.8.0.0/24 subnetwork will be routed through WireGuard Easy, hence the public IP does not change.
Note: this will not change value of `AllowedIPs` on existing peers (clients). This means that for `WG_ALLOWED_IPS` changes to take effect, for example, on phone, existing tunnel must be removed and re-added.

> An other thing that I've noticed, is that even having the following conf:
>
>  ```
>    --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
>    --sysctl="net.ipv4.ip_forward=1" \
>  ```
>
> my /etc/sysctl.conf file did not change

Those options are meant for WireGuard Easy container, they will not change the host's configuration.

**Comment by StefanoFranchini** [comment]
Hi @eandersons, I tried to reach my NAS (where WireGuard is also running) and an Arduino by ip address using the browser of my iPhone connected with WireGuard client app, but the connection with the server could not be established.

Each time I changed a config, I deleted the container, re-runnned the docker run command, created a new configuration on the GUI and scanned a new QR code to set up my client.

I don't have my own DNS server, this is why I always reach my local devices via IP address.

I tried:

`-e WG_ALLOWED_IPS=10.8.0.0/24`

cause my home network is 10.8.0.X, but also with the default 

`0.0.0.0/0, ::/0`

it won't work.

**Comment by eandersons** [comment]
Well, I am afraid that at this point I cannot help much, because I do not have an iPhone at hand to test with my WireGuard Easy server.

The last thing I can suggest to check to see if the issue is with server or client, is to ping your local network from within WireGuard Easy container: `docker container exec wg-easy ash`, then `ping x.x.x.x`.

Also, did you try to create a tunnel to your WireGuard Easy server using another device to see if the issue can be observed there as well?

**Comment by StefanoFranchini** [comment]
Yes, I also tried with a second iPhone and a computer. I can't ping my local network from any device.

As soon as I get back home (in some days) I will try to ping from the docker container (actually a good point, I hadn't think about that).

Thanks for you help by the way!

**Comment by enricoroselino** [comment]
yes, i also experience this... the VPN full tunnel feature and ICMP is working but cant reached by other devices using SSH with wireguard IPs (refused to connect error messsage). 

i think this doesnt work as expected like ordinary wireguard instalation, because this line is working fine on "native" install

`-e WG_POST_UP='iptables -t nat -A POSTROUTING -s subnet of clients -o eth0 -j MASQUERADE' \
-e WG_POST_DOWN='iptables -t nat -D POSTROUTING -s subnet of clients -o eth0 -j MASQUERADE' \`

**Comment by loveseatmedia** [comment]
Seeing the same issues, i can ping my systems in my remote network but i can't access any services, www, ssh, smb, etc...

**Comment by SKB-CGN** [comment]
I have the same problem, running it in Docker on TrueNAS.

**Comment by loveseatmedia** [comment]
I am going to guess something is screwy with the MTU settings, i trashed my docker and deployed another solution, but someone might look at what its doing to the MTU and test lowering it.

**Comment by SKB-CGN** [comment]
I was struggeling today in the morning and noticed, that my mobile provider had some network issues.

After this one was working, the connection to WG was slow and I have had many dropped/broken connections.

After setting MTU to 1384 on the client, I am totally satisfied with the speed and access.

Try it!

**Comment by dongwoo-lee** [comment]
i have same issue too. (ubuntu 20.05.x)

my parameter is..

sudo docker run -d \
--name=wg-easy \
-e WG_HOST=public ip \
-e PASSWORD='mypass' \
-e WG_DEFAULT_ADDRESS=192.168.1.22x \
-e WG_DEFAULT_DNS=168.126.63.1 \
-e WG_ALLOWED_IPS=0.0.0.0/0, ::/0 \
-e WG_POST_UP='' \
-v /home/adsoft/wg-easy:/etc/wireguard \
-p 51820:51820/udp \
-p 51821:51821/tcp \
--cap-add=NET_ADMIN \
--cap-add=SYS_MODULE \
--sysctl="net.ipv4.conf.all.src_valid_mark=1" \
--sysctl="net.ipv4.ip_forward=1" \
--restart unless-stopped \
weejewel/wg-easy



The client wanted to be assigned an IP starting from 192.168.1.221, so set -e WG_DEFAULT_ADDRESS=192.168.1.22x \.
IP allocation was completed successfully, but ping failed to be sent to the local network.

**Comment by fengwenhao-hyyd** [comment]
i have same issue too

this is my compose file, i run it on my mac(ip 172.16.90.244)
my home network has multiple NAT, so i use a cloud server to get public ip
i use frp to mapping my mac 51821/udp port to server's 51821/udp,
and use my server's public ip as WG_HOST
```yaml
version: '3'
services:
  wg-easy:
    image: weejewel/wg-easy
    container_name: wg-easy
    environment:
      - WG_HOST=cloudServerPubIP
      - PASSWORD=123456
      - WG_ALLOWED_IPS=10.23.23.0/24, 172.16.64.0/18
      - WG_DEFAULT_ADDRESS=10.23.23.x
    volumes:
      - ./wg-easy:/etc/wireguard
    ports:
      - "51820:51820/udp"
      - "58888:51821/tcp"
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    sysctls:
      - net.ipv4.conf.all.src_valid_mark=1
      - net.ipv4.ip_forward=1
    restart: unless-stopped
```
172.16.64.0/18 is my internal ip space

then i use my android device to test, it connect to the wg
i tried to access 10.23.23.1:51821 which is in the container, it's success, i can see login view of wg-easy on my android device.
then i try to access my nginx server on mac(172.16.90.244:80), and other port, but only get time out

in the wg-easy container, use curl to access 172.16.90.244:80 success, so the conneced of client -> wg-easy, wg-easy -> mac should be fine, did i miss some setting for iptables or others

**Comment by BobWs** [comment]
Having the same problem, wg connects fine from any client device but I cannot connect to my local network using my local domain names. In some occasions I can connect with the local IP e.g. DSFile can connect with the local IP of my Synology but not with the local subdomains.
I'm running Synology DNS-Server and also using docker NPM for reverse proxy.
Strangely enough when I connect to my local network using OpenVPN everything works fine, I can access my local network via IP and subdomains.

I could have sworn that at some point WG was working like it supposed to be but now I'm not sure what went wrong and why I stopped working properly.

**Comment by runopsrun** [comment]
> Having the same problem, wg connects fine from any client device but I cannot connect to my local network using my local domain names. In some occasions I can connect with the local IP e.g. DSFile can connect with the local IP of my Synology but not with the local subdomains. I'm running Synology DNS-Server and also using docker NPM for reverse proxy. Strangely enough when I connect to my local network using OpenVPN everything works fine, I can access my local network via IP and subdomains.
> 
> I could have sworn that at some point WG was working like it supposed to be but now I'm not sure what went wrong and why I stopped working properly.

I am fairly certain if you lower your MTU you can get it working. Not sure if its wg-easy at cause or a update with wireguard at cause.

**Comment by BobWs** [comment]
> I am fairly certain if you lower your MTU you can get it working. Not sure if its wg-easy at cause or a update with wireguard at cause.

I have set MTU to 1500, do you think this is sufficient? What value do you recommand?

**Comment by idf3da** [comment]
If anyone has this issue, set the keepalive to 24 sec since all routers reset the connections behind NAT after 25 sec.
Hope it helps someone. 🫡

**Comment by ivan-882** [comment]
> I am going to guess something is screwy with the MTU settings, i trashed my docker and deployed another solution, but someone might look at what its doing to the MTU and test lowering it.
You guys are my heroes.

@loveseatmedia @SKB-CGN

**Comment by pedroetb** [comment]
For me, the problem was a collision with real local network devices. After reading <https://blog.vforvruno.cyou/wireguard-local-network-unreachable>, I tried allowing specific address ranges (although may appear redundant, it's not when you have address collision).

At vpn client local network, I'm using `10.0.0.0/10` (10.0.0.1 to 10.63.255.254), and at server local network, I'm using `10.0.0.0/8` (10.0.0.1 to 10.255.255.254). So when trying to connect to `10.0.0.1` after setting up wireguard connection, I got response from router at client local network, instead of router at server local network.

After setting client configuration with `AllowedIPs = 0.0.0.0/0, ::/0, 10.0.0.0/10`,  connecting to `10.0.0.1` began to get response from router at server local network! Also works with masks greater than /10 (and lower than /32, obviously).

This wouldn't be necessary if I were using different IP ranges at both ends, but if you do you always can allow this traffic explicitly.

Also, tried and adopted other config values commented in this issue, like `MTU = 1400` and `PersistentKeepalive = 24`, but setting AllowedIPs made the difference at least.

**Comment by Ismola** [comment]
> For me, the problem was a collision with real local network devices. After reading https://blog.vforvruno.cyou/wireguard-local-network-unreachable, I tried allowing specific address ranges (although may appear redundant, it's not when you have address collision).
> 
> At vpn client local network, I'm using `10.0.0.0/10` (10.0.0.1 to 10.63.255.254), and at server local network, I'm using `10.0.0.0/8` (10.0.0.1 to 10.255.255.254). So when trying to connect to `10.0.0.1` after setting up wireguard connection, I got response from router at client local network, instead of router at server local network.
> 
> After setting client configuration with `AllowedIPs = 0.0.0.0/0, ::/0, 10.0.0.0/10`, connecting to `10.0.0.1` began to get response from router at server local network! Also works with masks greater than /10 (and lower than /32, obviously).
> 
> This wouldn't be necessary if I were using different IP ranges at both ends, but if you do you always can allow this traffic explicitly.
> 
> Also, tried and adopted other config values commented in this issue, like `MTU = 1384` and `PersistentKeepalive = 24`, but setting AllowedIPs made the difference at least.

I think it's a little late to say this, but I'm passing on my learning so that if anyone has problems, they can solve it.
After connecting to wg-easy, I did not have access to my local devices.
The solution:
`- WG_ALLOWED_IPS=0.0.0.0/0, 192.168.1.0/24`
// change 192.168.1.0/24 for the range of IPs you want to have access to within your local network. In my case, all my devices connect to 192.168.1.X

Next, you have to create a new wireguard client, because if you use one previously created for this change, the configuration will not update. I learned this from the website that my colleague shared

**Comment by phynias** [comment]
the only way to can connect to anything in my network is if i add 0.0.0.0/0 
if i have it just 192.168.1.0/24. i can ping but can't connect to anything inside.

is there anyway around this? i would much rather limit it.

**Comment by kolosek** [comment]
@Ismola Great it worked. An addition you can add to docker-compose.yml  

- WG_ALLOWED_IPS=192.168.1.0/24, 10.8.0.0/24

**Comment by kylmp** [comment]
For me lowering the MTU fixes it (1360), but I'd like to know why...

My problem specifically was accessing local devices on different (non 80 or 443) ports

**Comment by kylmp** [comment]
> [@kylmp](https://github.com/kylmp) Did you find a fix? Facing the same problem.

Try lowering the client MTU more until it works. 

Also, you can try to add the LAN for your wireguard's LAN as a value in client config allowedIps (i.e something like 192.168.0.0/24 or 10.0.0.0/24 - whatever your LAN is where the wireguard server is running)

Last, there could be collision if your local IP on your current network uses the same LAN IPs are your LAN on the wireguard's network. So, if your network where wireguard server is running is 10.0.0.0/24 and your local IP on your network you are trying to connect from is also a 10.0.0.XXX value it might have issues. So, you can test this by using a mobile network instead to see if this is the case (mobile network shouldn't have any collision)

**Comment by Paynayt** [comment]
>     2. If you only need to access a specific device on your local network (for example, 192.168.1.200), you can configure AllowedIPs as follows:
>        ```
>        AllowedIPs = 0.0.0.0/0, 192.168.1.200/32
>        ```

Thanks mate !! I guess wireguard wants to have the `0.0.0.0/0` ip to accept the nexts.

**Comment by sthames42** [comment]
**TL;DR**
Adding a static route to the LAN router, such that all traffic destined for `10.8.0.0/24` is sent to the LAN IP address of the VPN server, will allow devices on the LAN network to reply to the VPN client.

-----
After following [these instructions](https://www.wundertech.net/how-to-set-up-wireguard-on-truenas-scale/), I installed `wg-easy` on my TrueNAS-Scale server and found this page after experiencing the same problems as the OP.

- LAN router running ASUS/Merlin firmware and hosting DNS and DHCP servers.
- TrueNAS-Scale server running `wg-easy` WireGuard server:  
   ```
  WG_HOST=<WAN IP>
  WG_PORT=99999
  WG_DEFAULT_DNS=<LAN Router>
  WG_DEFAULT_ADDRESS=10.8.0.x
  WG_ALLOWED_IPS=<LAN Network>
  ```
- WAN port 99999 forwarded to LAN IP address of TrueNAS server.

From the VPN client, the TrueNAS server and `wg-easy` WebUI were accessible, but DNS would not work, and no other devices on the LAN would respond. This looked to me like a routing problem which is common to VPNs.

The VPN server is the gateway/router for the VPN network 10.8.0.x, just as the LAN router is the gateway/router for the LAN network. The VPN server has an IP address on both segments.

When I diagnosed the traffic patterns, it was clear the VPN client and server could talk to each other, since they were both on the `10.8.0.x` network, and the VPN server could route those VPN packets to other devices on the LAN network. But, those devices did not know how to reply to the VPN client, knowing of no route back to `10.8.0.x`. So, their replies were sent to the LAN router (their default route), which didn't know the route back to the VPN, either.

[The problem is well described, here](https://try.popho.be/vpn.html#my-vpn-server-is-not-the-router), and the solution of adding a static route pushed to each device on the LAN will work, but requires a change to the DHCP server and that all devices on the LAN get their IP address from DHCP--static IP assignments would require static route assignments.

It was much simpler to add a static route to the LAN router sending all packets destined for the `10.8.0.x` network to the LAN IP address of the VPN server. Once done, the `wg-easy` configuration, above, worked with no issues.

**Comment by ralfawilke** [comment]
I solved the issue "Can't connect to local devices." by allowing port 80 (for Webstation and Web Mail applications) in the Synology firewall. In my case it had nothing to do with the router.
